### ISSUE TITLE
Show Phone numbers based on a specific account only (Solved)

### ISSUE THREAD

**Issue by Nouradawy** [QuisApp/flutter_contacts#60]
i am trying to display all account phone numbers based on account type only as if you tried to display phone numbers for one contact you will find duplicates as there is no way to filter out the phone numbers,

contact.info?.phones[index].number

please I need to display only the numbers for the default viewing account only which is 0

**Comment by joachim-quis** [comment]
I'm not sure I completely understood the question. The problem is that a single contact has multiple/duplicate phone numbers? If you want to deduplicate them you can always to so manually, for example:

```dart
contact.phones.map((x) => x.number).toSet().toList()
```

Or if you only want the first phone:

```dart
contact.phones.first.number
```

If the question is about keeping only the phone numbers associated with a specific account (e.g. just the data from Gmail, or just the data from WhatsApp, etc.), this is currently not possible.

Let me know if that didn't answer the question.

**Comment by Nouradawy** [comment]
yes my Question was about keeping only the phone numbers associated with a specific account
and here is how I fixed it :
\flutter_contacts-1.1.5\android\src\main\kotlin\co\quis\flutter_contacts\FlutterContacts.kt
@line 282
```
 val phone = PPhone(
                                getString(Phone.NUMBER),
                                getString(RawContacts.ACCOUNT_TYPE),
                                getString(RawContacts.ACCOUNT_NAME),
                                getString(Phone.NORMALIZED_NUMBER),
                                label,
                                customLabel,
                                getInt(Phone.IS_PRIMARY) == 1
                            )
```
flutter_contacts-1.1.5\android\src\main\kotlin\co\quis\flutter_contacts\properties\Phone.kt
adding thous new variables to the maps

var AccountType: String,
 var AccountName: String,

fromMap and toMap
```
fun fromMap(m: Map<String, Any>): Phone = Phone(
            m["number"] as String,
            m["AccountType"] as String,
            m["AccountName"] as String,
            m["normalizedNumber"] as String,
            m["label"] as String,
            m["customLabel"] as String,
            m["isPrimary"] as Boolean
        )
    }


  fun toMap(): Map<String, Any> = mapOf(
        "number" to number,
        "AccountType" to AccountType,
        "AccountName" to AccountName,
        "normalizedNumber" to normalizedNumber,
        "label" to label,
        "customLabel" to customLabel,
        "isPrimary" to isPrimary
    )
```

finally updating it in the flutter side to receive
flutter_contacts-1.1.5\lib\properties\phone.dart

```
Phone(
    this.number, {
        this.AccountType= '',
        this.AccountName= '',
        this.normalizedNumber = '', 
        this.label = PhoneLabel.mobile, 
        this.customLabel = '', 
        this.isPrimary = false,
  });

  factory Phone.fromJson(Map<String, dynamic> json) => Phone(
        (json['number'] as String?) ?? '',
        'AccountType: (json['AccountType'] as String?) ?? '',
        AccountName: (json['AccountName'] as String?) ?? '',
        normalizedNumber: (json['normalizedNumber'] as String?) ?? '',
        label: _stringToPhoneLabel[json['label'] as String? ?? ''] ??
            PhoneLabel.mobile,
        customLabel: (json['customLabel'] as String?) ?? '',
        isPrimary: (json['isPrimary'] as bool?) ?? false,
      );

 Map<String, dynamic> toJson() => <String, dynamic>{
        'number': number,
        'AccountType': AccountType,
        'AccountName': AccountName,
        'normalizedNumber': normalizedNumber,
        'label': _phoneLabelToString[label],
        'customLabel': customLabel,
        'isPrimary': isPrimary,
      };

 int get hashCode =>
      number.hashCode ^
      AccountType.hashCode ^
      AccountName.hashCode ^
      normalizedNumber.hashCode ^
      label.hashCode ^
      customLabel.hashCode ^
      isPrimary.hashCode;

bool operator ==(Object o) =>
      o is Phone &&
      o.number == number &&
      o.AccountType == AccountType &&
      o.AccountName == AccountName &&
      o.normalizedNumber == normalizedNumber &&
      o.label == label &&
      o.customLabel == customLabel &&
      o.isPrimary == isPrimary;

  @override
  String toString() =>
      'Phone(number=$number,**AccountType=$AccountType ,AccountName=$AccountName,** normalizedNumber=$normalizedNumber, label=$label, '
      'customLabel=$customLabel, isPrimary=$isPrimary)';


```

**Comment by Nouradawy** [comment]
> yes my Question was about keeping only the phone numbers associated with a specific account and here is how I fixed it : \flutter_contacts-1.1.5\android\src\main\kotlin\co\quis\flutter_contacts\FlutterContacts.kt @line 282
> 
> ```
>  val phone = PPhone(
>                                 getString(Phone.NUMBER),
>                                 getString(RawContacts.ACCOUNT_TYPE),
>                                 getString(RawContacts.ACCOUNT_NAME),
>                                 getString(Phone.NORMALIZED_NUMBER),
>                                 label,
>                                 customLabel,
>                                 getInt(Phone.IS_PRIMARY) == 1
>                             )
> ```
> 
> flutter_contacts-1.1.5\android\src\main\kotlin\co\quis\flutter_contacts\properties\Phone.kt adding thous new variables to the maps
> 
> var AccountType: String, var AccountName: String,
> 
> fromMap and toMap
> 
> ```
> fun fromMap(m: Map<String, Any>): Phone = Phone(
>             m["number"] as String,
>             m["AccountType"] as String,
>             m["AccountName"] as String,
>             m["normalizedNumber"] as String,
>             m["label"] as String,
>             m["customLabel"] as String,
>             m["isPrimary"] as Boolean
>         )
>     }
> 
> 
>   fun toMap(): Map<String, Any> = mapOf(
>         "number" to number,
>         "AccountType" to AccountType,
>         "AccountName" to AccountName,
>         "normalizedNumber" to normalizedNumber,
>         "label" to label,
>         "customLabel" to customLabel,
>         "isPrimary" to isPrimary
>     )
> ```
> 
> finally updating it in the flutter side to receive flutter_contacts-1.1.5\lib\properties\phone.dart
> 
> ```
> Phone(
>     this.number, {
>         this.AccountType= '',
>         this.AccountName= '',
>         this.normalizedNumber = '', 
>         this.label = PhoneLabel.mobile, 
>         this.customLabel = '', 
>         this.isPrimary = false,
>   });
> 
>   factory Phone.fromJson(Map<String, dynamic> json) => Phone(
>         (json['number'] as String?) ?? '',
>         'AccountType: (json['AccountType'] as String?) ?? '',
>         AccountName: (json['AccountName'] as String?) ?? '',
>         normalizedNumber: (json['normalizedNumber'] as String?) ?? '',
>         label: _stringToPhoneLabel[json['label'] as String? ?? ''] ??
>             PhoneLabel.mobile,
>         customLabel: (json['customLabel'] as String?) ?? '',
>         isPrimary: (json['isPrimary'] as bool?) ?? false,
>       );
> 
>  Map<String, dynamic> toJson() => <String, dynamic>{
>         'number': number,
>         'AccountType': AccountType,
>         'AccountName': AccountName,
>         'normalizedNumber': normalizedNumber,
>         'label': _phoneLabelToString[label],
>         'customLabel': customLabel,
>         'isPrimary': isPrimary,
>       };
> 
>  int get hashCode =>
>       number.hashCode ^
>       AccountType.hashCode ^
>       AccountName.hashCode ^
>       normalizedNumber.hashCode ^
>       label.hashCode ^
>       customLabel.hashCode ^
>       isPrimary.hashCode;
> 
> bool operator ==(Object o) =>
>       o is Phone &&
>       o.number == number &&
>       o.AccountType == AccountType &&
>       o.AccountName == AccountName &&
>       o.normalizedNumber == normalizedNumber &&
>       o.label == label &&
>       o.customLabel == customLabel &&
>       o.isPrimary == isPrimary;
> 
>   @override
>   String toString() =>
>       'Phone(number=$number,**AccountType=$AccountType ,AccountName=$AccountName,** normalizedNumber=$normalizedNumber, label=$label, '
>       'customLabel=$customLabel, isPrimary=$isPrimary)';
> ```

but I don't know why when I try to get contact by the ID I get a crash all is working fine other than using 

`await FlutterContacts.getContact(contactID);`
